napalm_ios/ios.py

```diff
@@ -1134,129 +1136,225 @@ def bgp_time_conversion(bgp_uptime):
+            for item in parse_neighbors['patterns']:
+                match = item['regexp'].match(line)
+                if match:
+                    neighbor_data_entry.update(match.groupdict())
+                    if item['record']:
+                        neighbor_data.append(copy.deepcopy(neighbor_data_entry))
+                        for field in parse_neighbors['no_fill_fields']:
+                            try:
+                                del neighbor_data_entry[field]
+                            except KeyError:
+                                pass
+                    break
+        router_id = None
+        # check that we got a list of dicts
+        assert summary_data and isinstance(summary_data, list)
+        for entry in summary_data:
+            assert isinstance(entry, dict)
+            assert 'router_id' in entry
+            # select the first router_id value
+            if not router_id:
+                router_id = entry['router_id']
+            # check that every other value matches
+            assert entry['router_id'] == router_id
+        assert neighbor_data and isinstance(neighbor_data, list)
+        for entry in neighbor_data:
+            assert isinstance(entry, dict)
+        # check the router_id looks like an ipv4 address
+        try:
+            ipaddress.IPv4Address(router_id)
+            router_id = py23_compat.text_type(router_id)
+        except ipaddress.AddressValueError:
+            raise
+        # add parsed data to output dict
+        bgp_neighbor_data['global']['router_id'] = router_id
         bgp_neighbor_data['global']['peers'] = {}
-
-        cmd_neighbor_table = 'show ip bgp summary | begin Neighbor'
-        output = self._send_command(cmd_neighbor_table).strip()
-        # Cisco issue where new lines are inserted after neighbor IP
-        output = re.sub(RE_IPADDR_STRIP, r"\1", output)
-        for line in output.splitlines():
-            line = line.strip()
-            if 'Neighbor' in line or line == '':
+        for entry in summary_data:
+            # check that remote_addr looks sane
+            remote_addr = str(ipaddress.ip_address(entry['remote_addr']))
+            # want lower case afi
+            afi = entry['afi'].lower()
+            # check that we're looking at a supported afi
+            if afi not in supported_afi:
                 continue
-            fields = line.split()[:10]
-            peer_id, bgp_version, remote_as, msg_rcvd, msg_sent, table_version, in_queue, \
-                out_queue, up_time, state_prefix = fields
-            peer_id = peer_id.replace('*', '')
-
-            if '(Admin)' in state_prefix:
-                is_enabled = False
-            else:
+            # get neighbor_entry out of neighbor data
+            neighbor_entry = None
+            for n in neighbor_data:
+                if (n['afi'].lower() == afi) and \
+                        (str(ipaddress.ip_address(n['remote_addr'])) == remote_addr):
```

Comment:
I assume 'n' is a neighbor, can you make this a more meaningful name. Also use the napalm-base text_type instead of str for the string type).